bin/compile

```diff
@@ -50,11 +50,13 @@ install_jdk ${BUILD_DIR}
 
 export GRADLE_OPTS=${GRADLE_OPTS-"-Dorg.gradle.daemon=false"}
 
-if [ ! -d $CACHE_DIR ] ; then
-  mkdir $CACHE_DIR
-fi
+mkdir -p "$CACHE_DIR/.gradle"
 
 export GRADLE_USER_HOME=${GRADLE_USER_HOME:-$CACHE_DIR/.gradle}
+export GRADLE_PROJECT_CACHE_DIR="${GRADLE_PROJECT_CACHE_DIR:-$BUILD_DIR/.gradle}"
+if [ ! -d "$GRADLE_PROJECT_CACHE_DIR" ]; then
+  ln -s "$CACHE_DIR/.gradle" "$GRADLE_PROJECT_CACHE_DIR"
```

Comment:
I would suggest leaving a comment explaining why a symlink is necessary here. Something like:

```
By default gradle will write its cache in `$BUILD_DIR/.gradle`, and modifying this requires passing 
in a new argument, which would be complicated. By symlinking this directory to the cache, we 
ensure it will be cached by default.
```